### PR TITLE
Format Log API according to Cloud guidelines

### PR DESCRIPTION
**Pull request by pavelkalinnikov**
This addresses #985 with regards to formatting/commenting.
I recommend reviewing commit by commit to see the effects of individual guideline points.

### CODE REVIEW

File: trillian_log_api.proto [hunk 1]
```diff
@@ -219,60 +235,83 @@ message GetLeavesByHashRequest {
...
+    //  - `google.rpc.FAILED_PRECONDITION`: A conflicting entry is already
+    //    present in the log, e.g., same `leaf_index` but different `leaf_data`.
     google.rpc.Status status = 2;
 }
 
+// A leaf of the log's Merkle tree, corresponds to a single log entry. Each leaf
+// has a unique `leaf_index` in the scope of this tree.
 message LogLeaf {
-    // merkle_leaf_hash is over leaf data and optional extra_data.
+    // Output only. The hash over `leaf_data`.
```
Comment (pavelkalinnikov) [inline trillian_log_api.proto]: Apparently, `extra_data` is [not included](https://github.com/google/trillian/blob/7185394727abf33444773cc1bacd6a5d7c98ab11/server/log_rpc_server.go#L105) in the hash. Should it be?

File: trillian_log_api.proto [hunk 2]
```diff
@@ -219,60 +235,83 @@ message GetLeavesByHashRequest {
 }
 
 message GetLeavesByHashResponse {
-    // TODO(gbelvin) reply with error codes.
+    // TODO(gbelvin) reply with error codes. Reuse QueueLogLeaf?
```
Comment (pavelkalinnikov) [inline trillian_log_api.proto]: selfnit: `QueuedLogLeaf`.

**Comment by pavelkalinnikov** [comment]
@daviddrysdale Thanks for your review, will address it shortly. Correct, no code changes.

**Comment by gdbelvin** [comment]
This looks good to me.  Thanks for cleaning this up!

At some point - perhaps on the next API revision bump - change the directory structure as well

File: trillian_log_api.proto (hunk 2 above)
Comment (pavelkalinnikov) [reply trillian_log_api.proto]: Done.

**Comment by pavelkalinnikov** [comment]
ping

File: trillian_log_api.proto (hunk 1 above)
Comment (AlCutter) [reply trillian_log_api.proto]: No, this is just a way of associating informational metadata with the entry at submit time. If a personality wanted to commit to that data then, as you say, they'd make it part of the `leaf_value`.

File: trillian_log_api.proto [hunk 3]
```diff
@@ -14,75 +14,106 @@
...
+option java_package = "com.google.trillian.proto";
 
-import "trillian.proto";
-import "google/rpc/status.proto";
 import "google/api/annotations.proto";
 import "google/protobuf/timestamp.proto";
+import "google/rpc/status.proto";
+import "trillian.proto";
+
+// Provides access to a Verifiable Log as defined in the [Verifiable Data
```
Comment (Martin2112) [inline trillian_log_api.proto]: Can you tidy this comment while you're there. It says "provides access" and "provides direct access". Could be shortened.

**Comment by Martin2112** [comment]
This is another PR where GitHub does a terrible job of change tracking.

File: trillian_log_api.proto (hunk 3 above)
Comment (pavelkalinnikov) [reply trillian_log_api.proto]: Done.

**Pull request merged**